### PR TITLE
[Config][AWS]: Management of tyr parameters by env variables

### PR DESCRIPTION
**Pull request by azime**
Example for tyr instance :
**Old configration (fr.ini)**
image

**New configuration:**
`
TYR_INSTANCE_fr = {"instance":{"name":"fr","source-directory":"/srv/ed/source/fr","backup-directory":"/srv/ed/backup/fr/","aliases_file":"/usr/shared/ed/aliases_fr","synonyms_file":"/usr/shared/synonyms_fr","target-file":"/srv/kraken/exports/fr/data.nav.lz4","tmp_file":"/srv/ed/fr/datatmp.nav.lz4","is-free":false,"exchange":"exchange"},"database":{"host":"localhost1","dbname":"ed_fr","username":"ed_fr","password":"test","port":492}}`

### CODE REVIEW

File: source/tyr/tests/integration/helper_test.py [hunk 1]
```diff
@@ -54,3 +55,59 @@ def test_is_activate_autocomplete_version_without_mimir(disable_mimir):
...
+        str(exc.value) == 'Config is not valid for instance fr-se-lyon, error u\'492\' is not of type \'number\''
+    )
+
+
+def test_get_instances_name(init_instances_dir, valid_instance_env_variables):
+    with app.app_context():
+        instancies = get_instances_name()
+        assert len(instancies) == 2
+        for name in ["fr", "fr-se-lyon"]:
+            assert name in instancies
```
Comment (pbench) [inline source/tyr/tests/integration/helper_test.py]:
```suggestion
        instances = get_instances_name()
        assert len(instances) == 2
        for name in ["fr", "fr-se-lyon"]:
            assert name in instances
```

File: source/tyr/tests/integration/helper_test.py [hunk 2]
```diff
@@ -54,3 +55,59 @@ def test_is_activate_autocomplete_version_without_mimir(disable_mimir):
...
+
+
+def test_get_instances_name_same_instance(init_instances_dir, valid_instance_env_variables_fr):
+    # the same instance in config file and env variables
+    with app.app_context():
+        instancies = get_instances_name()
+        assert len(instancies) == 1
+        assert "fr" in instancies
```
Comment (pbench) [inline source/tyr/tests/integration/helper_test.py]:
```suggestion
        instances = get_instances_name()
        assert len(instances) == 1
        assert "fr" in instances
```

File: source/tyr/tyr/default_settings.py [hunk 3]
```diff
@@ -27,21 +28,21 @@
 CITIES_OSM_FILE_PATH = os.getenv('TYR_CITIES_OSM_FILE_PATH', '.')
 
 # Path to the directory where the configuration file of each instance of ed are defined
-INSTANCES_DIR = '.'
+INSTANCES_DIR = os.getenv('TYR_INSTANCES_DIR', '.')
 
 # Path to the directory where the data sources for autocomplete are stocked
-TYR_AUTOCOMPLETE_DIR = "/srv/ed/autocomplete"
+AUTOCOMPLETE_DIR = os.getenv('TYR_AUTOCOMPLETE_DIR', "/srv/ed/autocomplete")
```
Comment (pbench) [inline source/tyr/tyr/default_settings.py:34]: Is changing the name of this variable retro-compatible with the current deployment mechanism ? 
I am wondering if  fabric or something else is setting this variable. If this is the case, this change will break the deployment.

**Review by woshilapin — APPROVED** [review]

File: source/tyr/tyr/default_settings.py (hunk 3 above)
Comment (azime) [reply source/tyr/tyr/default_settings.py:34]: For me, no impact.
I checked the deployment script

**Review by pbench — APPROVED** [review]

**Pull request merged**